extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -692,22 +718,12 @@ export class AmpAnalytics extends AMP.BaseElement {
     const dynamicBindings = {};
     const resourceTimingSpec = trigger['resourceTimingSpec'];
     if (resourceTimingSpec) {
-      const on = trigger['on'];
-      if (on == 'ini-load') {
-        const binding = 'RESOURCE_TIMING';
-        const analyticsVar = 'resourceTiming';
-        // TODO(warrengm): Consider limiting resource timings to avoid
-        // duplicates by excluding timings that were previously reported.
-        dynamicBindings[binding] =
-            serializeResourceTiming(resourceTimingSpec, this.win);
-        expansionOptions.vars[analyticsVar] = binding;
-      } else {
-        // TODO(warrengm): Instead of limiting resource timing to ini-load,
-        // analytics should have throttling or de-dupe timings that have already
-        // been reported.
-        user().warn(
-            TAG, 'resource timing is only allowed on ini-load triggers');
-      }
+      const binding = 'RESOURCE_TIMING';
+      const analyticsVar = 'resourceTiming';
+      const after = this.getAndUpdateLastReportedTime_(resourceTimingSpec);
```

Comment:
I feel we can add some optimization here. In the case that `resourceTimingSpec['responseAfter']` is a large say 10000. We don't need to bind `RESOURCE_TIMING` when `performance.now()` is less than 10000. it saves the extra work to filter resource timing extras.  WDYT?

Reply:
I added a bit to exit early once the resource timing buffer has been fully read. WDYT?

Reply:
resource timing buffer is different.  I still think the best to support this is to check the current `performance.now()` value, and the initial `resourceTimingSpec['responseAfter']` value. If `performance.now() < resourceTimingSpec['responseAfter']`, do not bind resource_timing to expansionOptions.